crates/block-producer/src/errors.rs

```diff
@@ -67,9 +68,22 @@ pub enum AddTransactionError {
 /// queue can re-queue them.
 #[derive(Debug, PartialEq, Eq, Error)]
 pub enum BuildBatchError {
+    #[error(
+        "Too many input notes in the batch. Got: {0}, max: {}",
```

Comment:
Nit: I think `limit` is more precise than `max`, same below.
```suggestion
        "Too many input notes in the batch. Got: {0}, limit: {}",
```